roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -20,14 +20,14 @@ pub struct Amount(f64);
 pub struct BlockHash(Hash);
 
 #[derive(Clone, Debug)]
-pub struct TransacrtionWithHash {
-    id: Txid,
-    tx: Transaction,
+pub struct TransactionWithHash {
+    pub id: Txid,
+    pub tx: Option<Transaction>,
```

Comment:
What is the purpose of storing a txid without its corresponding full transaction?

Reply:
When you run the JDS, it starts creating its local mempool (through the use of a hashmap) by just loading the txids (which are the key of the hashmap).
As soon as `DeclareMiningJob` messages are received, JDS fills those entries by retrieving tx data.
In the meantime, it keeps on updating its local mempool, and the txids are used to retrieve (since are the hashmap keys) and pass tx data present in the current mempool to the updated one (so that it's no needed a second fetch from the node).
Here's the specific lines I'm mentioning: https://github.com/GitGab19/stratum/blob/93901fb4c525ccbedfe0dce7714ba11001d78ad1/roles/jd-server/src/lib/mempool/mod.rs#L86